### PR TITLE
Parse ChangeParams Action

### PR DESCRIPTION
**Pull request by majecty**
This PR depends on #10.
Only the last two commits are the new commits.

### CODE REVIEW

File: src/transactions.ts [hunk 1]
```diff
@@ -7,13 +7,15 @@ import {
 import { SDK } from "codechain-sdk";
 import { Custom } from "codechain-sdk/lib/core/transaction/Custom";
 const RLP = require("rlp");
+import { decode } from "punycode";
```
Comment (foriequal0) [inline src/transactions.ts]: Is punycode used?

**Review by foriequal0 — CHANGES_REQUESTED** [review]

File: src/transactions.ts [hunk 2]
```diff
@@ -174,6 +184,19 @@ export function actionFromRLP(sdk: SDK, rlp: Buffer): Action {
                 deposit: decodeU64(decoded[1]),
                 metadata: decoded[2]
             };
+        case CHANGE_PARAMS_ACTION_ID:
+            if (decoded.length <= 3) {
+                throw new Error(
+                    "A length of a RLP list of a changeParams action should be more than 3"
+                );
+            }
+            const [, , , signatures] = decoded;
```
Comment (foriequal0) [inline src/transactions.ts]: The layout of `decoded` is `[TAG, metadataSeq, params, sig1, sig2, sig3, ...]`
You need to triple-dot operator to assign all signatures to `signatures`
```js
const [, , , ...signatures] = decoded;
```
or just
```js
const signatures = decoded.slice(3);
```

**Review by foriequal0 — CHANGES_REQUESTED** [review]

**Comment by majecty** [comment]
@foriequal0 I fixed the PR.

File: src/transactions.ts [hunk 3]
```diff
@@ -174,6 +183,19 @@ export function actionFromRLP(sdk: SDK, rlp: Buffer): Action {
                 deposit: decodeU64(decoded[1]),
                 metadata: decoded[2]
             };
+        case CHANGE_PARAMS_ACTION_ID:
+            if (decoded.length <= 3) {
+                throw new Error(
+                    "A length of a RLP list of a changeParams action should be more than 3"
```
Comment (foriequal0) [inline src/transactions.ts:189]:
```suggestion
                    "A length of a RLP list of a changeParams action should be more than 4"
```

**Review by foriequal0 — DISMISSED** [review]

File: src/transactions.ts (hunk 3 above)
Comment (majecty) [reply src/transactions.ts:189]: Is it right to say "should be more than 3."
"x should be more than 3" == (x > 3)
"x should be more than 4" == (x > 4)

Comment (foriequal0) [reply src/transactions.ts:189]: Sorry! You're right. It was my mistake.

**Review by foriequal0 — APPROVED** [review]

**Comment by majecty** [comment]
@joojis Could you review this PR?

**Review by joojis — APPROVED** [review]

**Comment by joojis** [comment]
Sorry for the late review. I missed it.

**Pull request merged**